src/picongpu/include/particles/ionization/byCollision/ThomasFermi/AlgorithmThomasFermi.hpp

```diff
@@ -0,0 +1,150 @@
+ */
+
+namespace picongpu
+{
+namespace particles
+{
+namespace ionization
+{
+
+    /** \struct AlgorithmThomasFermi
+     *
+     * \brief calculation for the Thomas-Fermi pressure ionization model
+     *
+     * This model uses local density and "temperature" values as input
+     * parameters. To be able to speak of a "temperature" an equilibrium state
+     * would be required. Typical high power laser-plasma interaction is highly
+     * non-equilibrated, though. The name "temperature" is kept to illustrate
+     * the origination from the Thomas-Fermi model. It is nevertheless
+     * more accurate to think of it as an averaged kinetic energy.
+     */
+    struct AlgorithmThomasFermi
+    {
+
+        /** Functor implementation
+         *
+         * \tparam DensityType type of number
+         * \tparam KinEnergyDensityType type of kinetic energy density
+         * \tparam ParticleType type of particle to be ionized
+         *
+         * \param density number density value
+         * \param kinEnergyDensity kinetic energy density value
+         * \param parentIon particle instance to be ionized
+         */
+        template<typename KinEnergyDensityType, typename DensityType, typename ParticleType >
+        HDINLINE void
+        operator()( const KinEnergyDensityType kinEnergyDensity, const DensityType density, ParticleType& parentIon )
+        {
+
+            /* @TODO replace the float_64 with float_X and make sure the values are scaled to PIConGPU units */
+            const float_64 protonNumber = GetAtomicNumbers<ParticleType>::type::numberOfProtons;
+            const float_64 neutronNumber = GetAtomicNumbers<ParticleType>::type::numberOfNeutrons;
+            float_64 chargeState = attribute::getChargeState(parentIon);
+
+            /* ionization condition */
+            if (chargeState < protonNumber)
+            {
+                /* atomic mass number (usually A) A = N + Z */
+                float_64 massNumber = neutronNumber + protonNumber;
+
+                /** @TODO replace the static_cast<float_64> by casts to float_X
+                 * or leave out entirely and compute everything in PIConGPU scaled units
+                 */
+                float_64 const densityUnit = static_cast<float_64>(particleToGrid::derivedAttributes::Density().getUnit()[0]);
+                float_64 const kinEnergyDensityUnit = static_cast<float_64>(particleToGrid::derivedAttributes::EnergyDensity().getUnit()[0]);
+                /* convert from kinetic energy density to average kinetic energy per particle */
+                float_64 kinEnergy = (kinEnergyDensity / density) * (kinEnergyDensityUnit / densityUnit);
+                /** convert kinetic energy in J to "temperature" in eV by assuming an ideal electron gas
+                 * E_kin = 3/2 k*T
+                 */
+                float_64 temperature = kinEnergy * UNITCONV_Joule_to_keV * float_64(1.e3) * float_64(2./3.);
```

Comment:
What do we actually want to call an *"electron temperature"*?

@ax3l , @HighIander : ideas?

**ToDo:** Implement cut-off at 50 keV (?)  / Make the cut-off a user-definable parameter.